source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,406 @@
+
+Example
+^^^^^^^
+
+This example shows how to make a listener class that prints a message each time
+you check out a connection from your connection pool.
+
+The following code defines the ``ConnectionPoolLibrarian`` class which implements the
+``ConnectionPoolListener`` interface.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Monitoring.java
+   :language: java
+   :dedent:
+   :start-after: start cp-listener-impl
+   :end-before: end cp-listener-impl
+
+The following code adds an instance of the ``ConnectionPoolLibrarian`` class to a
+``MongoClient`` object. The code then runs a database command to test the
+librarian.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Monitoring.java
+   :language: java
+   :dedent:
+   :start-after: start monitor-cp-example
+   :end-before: end monitor-cp-example
+
+The above code snippet should produce output that looks like this:
+
+.. code-block:: none
+   :copyable: false
+
+   Let me get you the connection with id 21...
+   Let me get you the connection with id 21...
+
+It looks like our find operation requested two connections from the pool.
+If you are curious why the driver uses two connections in one find operation, 
+`you can find the answer in this line of the MongoDB driver source code <https://github.com/mongodb/mongo-java-driver/blob/a82ce71d0aa7da9c6455f1b52ae24ed3cd4c1f62/driver-sync/src/main/com/mongodb/client/internal/FindIterableImpl.java#L203>`__.
+
+For more information on the classes and methods mentioned in this section, see
+the following API documentation:
+
+- :java-docs:`ConnectionPoolListener <apidocs/mongodb-driver-core/com/mongodb/event/ConnectionPoolListener.html>`
+- :java-docs:`MongoClientSettings <apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html>`
+- :java-docs:`MongoClient <apidocs/mongodb-driver-sync/com/mongodb/client/MongoClient.html>`
+- :java-docs:`ConnectionCheckedOutEvent <apidocs/mongodb-driver-core/com/mongodb/event/ConnectionCheckedOutEvent.html>`
+- :java-docs:`ConnectionCheckOutFailedEvent <apidocs/mongodb-driver-core/com/mongodb/event/ConnectionCheckOutFailedEvent.html>`
+
+.. _monitoring-jmx:
+
+Monitor Connection Pools with JMX
+---------------------------------
+
+You can monitor connection pool events using **Java Management Extensions (JMX)**. 
+JMX provides tools to monitor applications and devices.
+
+For more information on JMX, see 
+`the official Oracle JMX documentation <https://docs.oracle.com/javase/tutorial/jmx/index.html>`__. 
+
+JMX Support in the MongoDB Driver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable JMX connection pool monitoring, add an instance of the
+``JMXConnectionPoolListener`` class to your ``MongoClient``.
+
+The ``JMXConnectionPoolListener`` class performs the following actions: 
+
+- Creates MXBean instances for each ``mongod`` or ``mongos`` process the driver
+  maintains a connection pool with.
+- Registers these MXBean instances with the platform MBean server. 
```

Comment:
```suggestion
- Registers the MXBean instances with the platform MBean server. 
```

Reply:
I believe "these" is correct here. 

Definition of [these](https://www.google.com/search?q=these&oq=these&aqs=chrome..69i57j69i60l5j69i65j69i64.502j0j7&sourceid=chrome&ie=UTF-8): "referring to a specific thing or situation just mentioned."

Is there a reason why "the" is more appropriate here? Am I missing the crux of this suggestion?

Reply:
Sounds good. 

Reasons for keeping it as is:

I don't think this should be numbered, as in code this all happens as one function call, but I think it should be "these" as it is referencing the items created in the previous bullet.